CS483/CS483/Kartaclysm/Components/ComponentTrackPiece.cpp

```diff
@@ -154,6 +197,66 @@ namespace Kartaclysm
 		}
 	}
 
+	bool ComponentTrackPiece::IsOffroadAtPosition(glm::vec3 p_pPosition)
+	{
+		glm::vec3 checkPosition = p_pPosition;
+		checkPosition = checkPosition - m_pGameObject->GetTransform().GetTranslation();
+		checkPosition = checkPosition * -m_pGameObject->GetTransform().GetRotation();
+
+		float x = checkPosition.x;
+		float z = checkPosition.z;
+
+		x /= m_fWidthX;
+		z /= m_fWidthZ;
+
+		x *= 2.0f;
+		z *= 2.0f;
+
+		for (int i = 0; i < m_vOffroadSquares.size(); i++)
+		{
+			OffroadSquare square = m_vOffroadSquares[i];
+
+			if (((x >= square.x1 && x <= square.x2) || (x >= square.x2 && x <= square.x1))
+				&& ((z >= square.z1 && z <= square.z2) || (z >= square.z2 && z <= square.z1)))
+			{
+				return true;
+			}
+		}
+
+		for (int i = 0; i < m_vOffroadTriangles.size(); i++)
```

Comment:
Unsigned int mismatch. You should also look at using an iterator or for (auto triangle : m_vOffroadTriangles)